src/citrine/resources/ara_job.py

```diff
@@ -0,0 +1,179 @@
+    """
+    def __init__(self, job_id: UUID):
+        self.job_id = job_id
+
+
+class TaskNode(Resource['TaskNode']):
+    """
+    [ALPHA] individual task status
+
+    Parameters
+    ----------
+    id: str
+        unique identification number for the job task.
+    task_type: str
+        the type of task running
+    status: str
+        the last reported status of this particular task.
+    dependencies: Set[str]
+        all the tasks that this task is dependent on.
+    failure_reason: Optional[str]
+        if a task has failed, the failure reason will be in this parameter.
+    """
+    id = properties.String("id")
+    task_type = properties.String("task_type")
+    status = properties.String("status")
+    dependencies = PropertySet(String(), "dependencies")
+    failure_reason = properties.Optional(String(), "failure_reason")
+
+    def __init__(
+            self,
+            id: str,
+            task_type: str,
+            status: str,
+            dependencies: Set[str],
+            failure_reason: Optional[str]
+    ):
+        self.id = id
+        self.task_type = task_type
+        self.status = status
+        self.dependencies = dependencies
+        self.failure_reason = failure_reason
+
+
+class TaskNodeType(Property[TaskNode, dict]):
+    @property
+    def underlying_types(self):
+        return TaskNode
+
+    @property
+    def serialized_types(self):
+        return dict
+
+    def _serialize(self, value: TaskNode) -> dict:
+        result = {"id": value.id, "task_type": value.task_type, "status": value.status,
+                  "failure_reason": value.failure_reason, "dependencies": value.dependencies}
+        return result
+
+    def _deserialize(self, value: dict) -> TaskNode:
+        return TaskNode.build(value)
+
+
+class JobStatusResponse(Resource['JobStatusResponse']):
+    """
+    [ALPHA] a response to a job status check
+
+    Parameters
+    ----------
+    job_type: str
+        the type of job for this status report
+    status: str
+        the actual status of the job
+    tasks: List[TaskNode]
+        all of the constituent task required to complete this job
+    """
+
+    job_type = properties.String("job_type")
+    status = properties.String("status")
+    tasks = properties.List(TaskNodeType, "tasks")
+
+    def __init__(
+            self,
+            job_type: str,
+            status: str,
+            tasks: List[TaskNode]
+    ):
+        self.job_type = job_type
+        self.status = status
+        self.tasks = tasks
+
+
+class AraJobFramework(Resource["AraJobFramework"]):
+    """
+    [ALPHA] a resource for submitting a job and getting job status
+
+    Parameters
+    ----------
+    session: Session
+        the remote session for hitting framework endpoints
+    """
+    _path_template = 'projects/{project_id}'
+
+    def __init__(self, session: Session):
+        self.session = session
+
+    def build_ara_table(self, project: Project,
```

Comment:
I think I would expect this method to be a peer to https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/resources/ara_definition.py#L231 rather than living in this new `AraJobFramework`. That will also save you the trouble of figuring out how the user is supposed to actually create an instance of this class, too. @maxhutch were you imagining that's where this method would go? What about the job status?